components/purchase_modal/purchase_modal.tsx

```diff
@@ -200,28 +206,36 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
 
     listPlans = (): JSX.Element => {
         const products = this.props.products!;
-        const flatFeeProducts: ProductOptions = [];
-        const userBasedProducts: ProductOptions = [];
+        let flatFeeProducts: ProductOptions = [];
+        let userBasedProducts: ProductOptions = [];
         Object.keys(products).forEach((key: string) => {
-            const tempEl: Option = {key: products[key].name, value: products[key].id, price: products[key].price_per_seat};
+            const tempEl: RadioGroupOption = {key: products[key].name, value: products[key].id, price: products[key].price_per_seat};
             if (products[key].billing_scheme === BillingSchemes.FLAT_FEE) {
                 flatFeeProducts.push(tempEl);
             } else {
                 userBasedProducts.push(tempEl);
             }
         });
 
-        let options = [...flatFeeProducts.sort((a: Option, b: Option) => a.price - b.price), ...userBasedProducts.sort((a: Option, b: Option) => a.price - b.price)];
-
-        const currentPrice = this.state.currentProduct?.price_per_seat;
+        const currentProduct = this.state.currentProduct!;
 
         // if not on trial, only show current plan and those higher than it in terms of price
-        if (!this.props.isFreeTrial && currentPrice) {
-            options = options.filter((option: Option) => {
-                return option.price >= currentPrice;
+        if (!this.props.isFreeTrial && currentProduct.price_per_seat) {
+            userBasedProducts = userBasedProducts.filter((option: RadioGroupOption) => {
+                return option.price >= currentProduct.price_per_seat;
+            });
+            flatFeeProducts = flatFeeProducts.filter((option: RadioGroupOption) => {
+                return option.price >= currentProduct.price_per_seat;
```

Comment:
Is this suppose to be the `else` of the above `if (!this.props.isFreeTrial && currentProduct.price_per_seat)` ? What happens if it's flat_fee price?

Reply:
ok!